rerun_py/rerun_bindings/rerun_bindings.pyi

```diff
@@ -927,3 +930,341 @@ def asset_video_read_frame_timestamps_nanos(
+
+class DataframeQueryView:
+
+    def filter_partition_id(self, partition_id: str, *args: Iterable[str]) -> Self:
+        """Filter by one or more partition ids. All partition ids are included if not specified."""
+
+    def filter_range_sequence(self, start: int, end: int) -> Self:
+        """
+        Filter the view to only include data between the given index sequence numbers.
+
+        This range is inclusive and will contain both the value at the start and the value at the end.
+
+        The view must be of a sequential index type to use this method.
+
+        Parameters
+        ----------
+        start : int
+            The inclusive start of the range.
+        end : int
+            The inclusive end of the range.
+
+        Returns
+        -------
+        RecordingView
+            A new view containing only the data within the specified range.
+
+            The original view will not be modified.
+
+        """
+
+    def filter_range_secs(self, start: float, end: float) -> Self:
+        """
+        Filter the view to only include data between the given index values expressed as seconds.
+
+        This range is inclusive and will contain both the value at the start and the value at the end.
+
+        The view must be of a temporal index type to use this method.
+
+        Parameters
+        ----------
+        start : int
+            The inclusive start of the range.
+        end : int
+            The inclusive end of the range.
+
+        Returns
+        -------
+        RecordingView
+            A new view containing only the data within the specified range.
+
+            The original view will not be modified.
+
+        """
+
+    def filter_range_nanos(self, start: int, end: int) -> Self:
+        """
+        Filter the view to only include data between the given index values expressed as seconds.
+
+        This range is inclusive and will contain both the value at the start and the value at the end.
+
+        The view must be of a temporal index type to use this method.
+
+        Parameters
+        ----------
+        start : int
+            The inclusive start of the range.
+        end : int
+            The inclusive end of the range.
+
+        Returns
+        -------
+        RecordingView
+            A new view containing only the data within the specified range.
+
+            The original view will not be modified.
+
+        """
+
+    def filter_index_values(self, values: IndexValuesLike) -> Self:
+        """
+        Filter the view to only include data at the provided index values.
+
+        The index values returned will be the intersection between the provided values and the
+        original index values.
+
+        This requires index values to be a precise match. Index values in Rerun are
+        represented as i64 sequence counts or nanoseconds. This API does not expose an interface
+        in floating point seconds, as the numerical conversion would risk false mismatches.
```

Comment:
that's not _entirely_ true anymore though? we have now three distinct kinds of time one of them being nanoseconds since epoch

Reply:
Well, this api takes `i64` and ultimately tests against `TimeInt`. If I understand correctly, the seq/duration/datetime semantic layer is entirely short-circuited. So you could say that the mention of "nanoseconds" accurately "hand waves" the fact that this might be either relative or absolute since epoch.